### ISSUE TITLE
Allow the app to continue in background even when swiped away.

### ISSUE THREAD

**Issue by ThoriumTextile** [Kunzisoft/KeePassDX#1334]
Hello.
I often tend to close all background apps to free up memory, but with KPDX, doing so closes the database.
AFAIK there's no setting to allow the app to daemon-ify.

**Comment by J-Jamet** [comment]
This is what the application did several versions ago, but I was asked to do the opposite, which seemed to make more sense because the user is unequivocally asking to kill the application.

Maybe the easiest way is to put a setting in the parameters to let the user choose what is best for him.

**Comment by ThoriumTextile** [comment]
That's exactly what i mean, if you or some other contributor were to create a PR to implement this setting, it would be fantastic.

**Comment by J-Jamet** [comment]
Planned for version 4.0.0 as it will require structural changes.

**Comment by kraoli** [comment]
> Hello. I often tend to close all background apps to free up memory, but with KPDX, doing so closes the database. AFAIK there's no setting to allow the app to daemon-ify.

May I ask kindly about your use-case? And is it just a habbit to close all apps? I don't understand the benefit when having automatic memory management.  Do you leave keepass unlocked all the time? 

I'm one of those people who thinks features like leaving it unlocked by a background deamon are very dangerous in a sec product. Even as an option.

**Comment by terminal-root** [comment]
This would be a nice option to have, but it should definitely be opt-in in my opinion. I have my databases set to lock after a given time, but I don't want to have to keep the app open in the foreground during that time to avoid having to unlock them every time I fill creds.

I understand why some may think it counter-intuitive -- security-wise, at least -- to have your database open in the background indefinitely, and I agree. That is why I have my databases set to lock after some amount of time. Also, -- regarding the user telling the app to close -- after the user opts-in to the setting, they can manually lock a database in-app if they wish to do so early.

I recently made the switch from 1password to a self-hosted solution using KeePass databases, and I am really liking KeePassDX/XC. Thank you all for your contribution to the project!

**Comment by mahescho** [comment]
I want to use KeepassDX but the current behavior prevents me from doing so and I stick to Keepass2Android.

I use a Yubikey in conjunction with a password for my database which is shard on many devices by NextCloud. On Linux I use KeepassXC. Both, KeepassXC and Keepass2Android allow me to keep the DB open. With Keepass2Android I unlock my DB with my password and the Yubikey. I can setup Keepass2Android so I can quit it and reopen the DB with fingerprint.

With KesspassDX I can unlock by fingerpint but in addition I always need my Yubikey. This major annoyance keeps me away from KeepassDX.

**Comment by J-Jamet** [comment]
It's always a balance between security and usability, and here the behavior won't be changed by default, because killing the application with a swipe clearly means closing the database. I'm thinking about implementing the setting to prevent the base from closing during the swipe, but this requires a lot of testing.

In the case of Yubikey, this is a strong security mode. It's very surprising to leave your database open when you're using a very strong security mode. In any case, you can change the timeout in the settings.

**Comment by mahescho** [comment]
My DB resides on a network storage on the internet. This is the reason for using the Yubikey. The DB needs to be protected while on the internet storage. On my devices I neither want nor need this. The trade off is the one time unlocking with the Yubikey. When I'm at e.g. at home my phone is always with me, my Yubikey is't. When I need a password in this case I need to go and get my Yubikey to unlock. For me a major annoyance which prevents me from using the in many other aspects far better KeepassDX. With Keepass2Android this issue does not exist. 

Keepass2Android has an other weakness. The DB is writable without the Yubikey. For me KeepassXC offers the best balance of security and usability. With KeepassXC I need to tap my Yubikey when opening the DB. Then it stays open for the login session, no matter if the screen gets locked or not, but when I want to write to the DB I've to tap the Yubikey again.

**Comment by ColCh** [comment]
Same thing, it's a bit frustrating to keep yubikey with me even at home, and app seem to be to close db (or, its android closing the app) per 8h. 

It will be really nice to have a possibility to daemonize app / keep it it memory without closing

Sorry, may be I don't understand something, but android these days is secure enough to keep memory of some app separated enough from other apps? I mean, is it even possible to read other process memory on stock android, without root?

**Comment by J-Jamet** [comment]
Normally, each application is in a kind of Sandbox, unless the corresponding permissions allow to see outside. With root you can do anything. If you need more information, check out the AOSP code.

**Comment by ColCh** [comment]
sorry, I was not explicit enough. That was a comment about this statement above:

> In the case of Yubikey, this is a strong security mode. It's very surprising to leave your database open when you're using a very strong security mode. In any case, you can change the timeout in the settings.

soo, it's not completely clear to me, why it's surprising... I have to apply biometric finger to unlock the device, my phone's ROM is secure enough to keep another apps away from KepassDX memory (I have stock android, no root), my filesystem is encrypted, I'm pretty confident that nobody will use my PIN/Fingerprint to steal my data :-)

I already have unlock timeout set to 2 days (this is comfortable interval for me), but KeepassDX forces to reopen it every 4-8h, and forces me to unlock db with yubikey also, which is forcing me to wake up from my sofa and go search for my yubikey at my table O_O sorry, this may sound funny, but it's true... 

However, physical key emulator feature sounds really hard to implement... what about just ensuring that app will be opened for desired time (e.g. providing a switch in settings, et cetera)? Of course I'm not the one to suggest anything, or point on priorities (it's your project, I'm just using it / supporting it), I'm just asking.

P.S. Please note, I already have all Battery Optimizations set to OFF, for KeepassDX, so I'm 90% sure it's not android killing the app (never say 100% though)

**Comment by J-Jamet** [comment]
>  but KeepassDX forces to reopen it every 4-8h

There's nothing in the code of KeePassDX to force this.

> and forces me to unlock db with yubikey also, which is forcing me to wake up from my sofa and go search for my yubikey at my table O_O sorry

Haha, that's the whole point of choosing a physical key unlocking system, so that it requires human intervention.

> However, physical key emulator feature sounds really hard to implement...

True, but on the right track

> what about just ensuring that app will be opened for desired time (e.g. providing a switch in settings, et cetera)?

It's the case, maybe there's a protection on your device that prevents foreground services from running too long.

**Comment by mahescho** [comment]
Perhaps it may be an option to handle it like Keepass2Android, just give it a try to see how it works there.

This missing feature prevents me form using KeepassDX and I've keep using Keepass2Android.

**Comment by ColCh** [comment]
> It's the case, maybe there's a protection on your device that prevents foreground services from running too long.

I'm pretty sure there is no such thing, as I disabled all possible battery optimizations for keepassdx

however, I use foldersync to copy kdbx file from one device to another, may it unload the app because of file sync?

I don't understand, app should work without any limit ...

**Comment by rawpie2** [comment]
It should be when I swipe the app away it stays open too

**Comment by J-Jamet** [comment]
If you reject someone, don't expect them to be open to you afterwards :p. It's the same thing here, you're clearly indicating by the action of rejecting the application that you don't want to use it anymore, so why should it remain open?

**Comment by ColCh** [comment]
@J-Jamet any reason why key cannot be saved under biometric protection? I mean, it's possible to not completely lock DB even if app if closed. I totally understand why and how YUBI is used like it's used, but reopening db with yubikey every day is too much for me: for example, I leave DB opened in KeepassXC on my mac, I just need to tip my finger to finger scanner, and that's it, no need to use yubikey. 

Why there is no way to use it like this on android? I'm sure app is never closed by android system itself: I see that resilio sync is opened for 132h, for example. May be it's possible just to have a toggle inside of KeepassDX to force android system not to close the app? using some service, for example, I'm not sure (i'm web dev, not android dev)

**Comment by curable-online** [comment]
Hey, a happy user of KeePassDX.  Thank you for your application.  It's the best on Android.  I would donate regularly if I wasn't from a restricted country (just to emphasis that your app deserves user's support and if I don't support it in a particular way it's only because I really can't do so.).

But on the issue.  I think, you are wrong.  Swiping an app away just means I don't want to access it for now.  I swipe my Firefox away, when I don't plan to open it any more, but if it's downloading something, it keeps doing it until the download is done!

Swiping away doesn't necessarily mean to stop an app's background activity. Many Android apps still do stuff in the background, even though they are not in Recents screen.

Point is, an application can run in the background, and if it's user's desire, then it's the right thing to do so.  Is KeepPassXC's window always open or even minimized when a database is open?  Answer is no (databases stay open even when KeePassXC is in the background without any of its windows being open)!  Pretty much most other KeePass apps that I know on Window, Linux and Android don't hard bind Database's close/open state to the state of the app being in foreground, minimized or in background.  Of course, in Window and GNU/Linux, they show a tray icon, which is similar to how an app has to show a persistent notification do the similar thing on Android.

Not wanting KeePassXD to be in my Recents screen, isn't semantically hard bound to my desire for my databases to be closed or open.  If it is, then what is the point of the timer?  If all apps being open in the background would need to stay in the recent screen, then my recents screen would be filled with ten of them!

**Comment by curable-online** [comment]
And, one more problem is when KeePassDX opens up from another app through the MagiKeyboard.

When you want to enter your password in an application using Magikeyboard, you may need to tap the Magikeyboard's key button to select an entry from the list.  Then, when you are done doing that KeePassDX loses its own window from the Recents screen.  In the next step, you are done with your application where you entered your password in and want to swipe it away, if you don't plan to use it any more.  And tada, keepassDX is closed and you have to reopen your database!

This one is more look like a bug, because is never explicitly swiped the keepassdx away.

**Comment by curable-online** [comment]
> I'm pretty sure there is no such thing, as I disabled all possible battery optimizations for keepassdx
> 
> I don't understand, app should work without any limit ...

Battery optimization (technically called 'App Standby') is not the only thing which can close an app.  There is also the Doze mode (also called Idle mode) functionality which stops all background activities when your phone's screen is off and your phone is idle and not charging.  It's not manageable without root, as far as I know.

I don't make it up, please have a look at:

https://developer.android.com/training/monitoring-device-state/doze-standby

Which documents this.


However, it's behaving differently on various OEMs.  And the strict restriction to stop all network and background activity wasn't up until the Android's 13 version.

**Comment by J-Jamet** [comment]
That's what you think is the most appropriate behavior because you're used to it with other applications, but as I said at the beginning, this behavior of not closing the database was present at the beginning and was changed for very good reasons.

A user action inevitably has a consequence. KeePassDX works very simply: the database is only opened in RAM from the database service. In this desired behavior, the user specifically wants to delete as much RAM memory as possible, so that is what is done.
Explicitly asking the system to delete its backstack data through an action while simultaneously asking it to retain data as much as possible to keep the database open is counterintuitive and requires forcing the system.

You're just not used to it because other applications SEEM not to work that way, but using your example with Firefox, KeePassDX works technically the same way. When downloading a file, the database will only be closed once the high-priority action currently in progress has been completed. If you have a very large file to upload as attachment or the database has a large KDF and it takes a long time, you will see this.

When firefox is swiped from the backstack, it does not keep the DOM of the current page in memory; it only keeps the URL, and the entire page is reloaded based on the URL.
When KeePassDX is swiped, the database closes, and if you want to reopen it, you go through the file URI to decrypt the content and reload the XML. You can see that the concepts are identical, the difference being that you don't have to authenticate yourself each time on every web page. And KeePassDX ensures that no sensitive information is accessible when the database is closed.

If you want to compare it to how KeePassXC behaves on a PC, it's like close the program off the desktop when you make a global programm preview. KeePassXC is completely closed with the databases closed. In any case, the systems are not technically the same.

If you want to keep it running in the background, why not just keep the UI for back tasks open instead of closing it?
The system's "clear all" button simply has no equivalent on a desktop computer OS, but would have the same effect.

You mention recent screens several times, but these are not just recent screens; they indicate that programs have retained data in the backstack for later use. In layman's terms, this means that a task is running in the background and that deleting this task implies that the user no longer wants to use this application, so the data can be deleted.

The timer is only used to send a Lock event after a predetermined time set by the user. These are two different functions, so I don't understand what you mean here.

Same with Magikeyboard, I don't understand the problem. 
If you don't plan to using it anymore, then it's normal for the database to close. Otherwise, open another issue for this bug.

I believe it is simply a matter of habit but you say I'm wrong and you are not satisfied with the current behavior of the application, despite of KeePassDX was specifically designed to be consistent in this architecture, and reverting to the old behavior will cause workflow issues. 
Perhaps KeePassDX simply does not meet your needs, simply use another app if it bothers you that much.

I am willing to look into this feature when refactoring the code for version 5.0.0, but you should bear in mind that if it complicates things too much, I won't waste time on it because it's a  just an accomodation feature.

**Comment by curable-online** [comment]
> That's what you think is the most appropriate behavior because you're used to it with other applications,

Other applications AND KeePassDX, until it changed.

> A user action inevitably has a consequence.

Yes, and the consequence is ‘clearing your Recents screen’.  This UI is designed to help you switch among the tasks you was recently on.  It’s much like the taskbar in MS Windows.

>  to retain data as much as possible to keep the database open is counterintuitive and requires forcing the system

Not counter-intuitive for the user.  The previous behaviour of the app wasn’t counter-intuitive.  At least not for most users.

>  the user specifically wants to delete as much RAM memory as possible, so that is what is done.

I don't think if an average user thinks of the Clear Button as a RAM Cleaner.  Nothing about RAM and ‘RAM Cleaning’ is shown in the standard Recents/Overview screen.

I can make a long list of apps that stay in the background doing their job after being swiped and a longer list of the apps that don’t.

I don’t simply see the convention that you are trying to make as: ‘Swiping an app away MUST close the app, and MUST stop its functionality, and MUST clear it from the RAM.’.

Rather I think: ‘Swiping an app away MAY close the app, and MAY stop its functionality and, MAY clear it from the ram, *IF it’s user’s desire*!’

Now, at the moment, there is no consideration about ‘what the user may want!’.

In MS Windows, taping the X (close) button on top bar of the window of an application MAY close it, or MAY minimized it as a tray icon.  Even apps like VLC have that option.  You can further even hide the tray icon of an app from the task bar and put it in a drop up menu.

Now, about the Firefox, you are completely right.  But, my example was trying to say this: “Swiping an app away doesn’t necessarily mean that it has to stop doing something a user wants it to do.”

> If you want to compare it to how KeePassXC behaves on a PC, it's like close the program off the desktop when you make a global programm preview. KeePassXC is completely closed with the databases closed. In any case, the systems are not technically the same.

As far as I know, closing the KeePassXC window through the top right close button or from the taskbar (or from the overview in Gnome) minimize it to tray icons (if the system have them) and keeps the program running in the background with the state of the database in this condition being controllable through an option in the preferences.

> If you want to keep it running in the background, why not just keep the UI for back tasks open instead of closing it?  The system's "clear all" button simply has no equivalent on a desktop computer OS, but would have the same effect.

If you close the window/UI of all the programs in the desktop, some of them stay running and working in the background and some of them not.  Just like Android; clearing the Recents/Overview screen will stop some apps, while some of them stay running in the background (for them it is required to have a persistent notification which acts like the tray icons in the desktop).

> The timer is only used to send a Lock event after a predetermined time set by the user. These are two different functions, so I don't understand what you mean here.

One may have more than a hundred of apps installed, all of them are going to be used one day or the other, should all of them stay in the Recents/Overview screen?  How many hours or minutes or days has to be the idle time distance between two sessions of an app being used, for it to stay in the Recents?  Who defines it?  Is there an instruction about how to use the Overview screen?

> The timer is only used to send a Lock event after a predetermined time set by the user. These are two different functions, so I don't understand what you mean here.

The timer is not much useful if the app doesn’t stay open in the background.  I may set it to eight hours, should I keep it in the Overview screen for eight hours?  Isn’t it that the Overview has to contain apps that are used **Recently**.  How many hours is recent?  Is an app used yesterday considered **Recent**?  If yes, then most of the installed apps has to stay in Overview screen.

> I believe it is simply a matter of habit

I’m certainly here because something irritated me for a while; I may be wrong or may be right, but one would never know unless though discussion.

I have tried to use the credentials instead, but the KeePassDX doesn’t keep the key file.  So, I need to re-provide the key file each time.  What is the point of using the phone’s credentials then?  If I’m going to provide the key file, I would simply insert the password, too, which is easier than providing the keyfile.

> but you say I'm wrong, and you are not satisfied with the current behavior of the application,

I don’t think if the application is doing something wrong, if it’s the intended behaviour of the app by the developer.  So, it’s a feature request.  Not a bug report.

You may reject the request, and still you are doing everyone a favour by making a free (as beer) and free (as speech) program.

However, I use the right to disagree with your presumption about how the Overview screen SHOULD be used, as I described.

> Perhaps KeePassDX simply does not meet your needs, simply use another app if it bothers you that much.

It doesn’t meet one of my needs, indeed.  But still, it provides features not provided by other apps.

> I am willing to look into this feature when refactoring the code for version 5.0.0, but you should bear in mind that if it complicates things too much, I won't waste time on it because it's a just an accomodation feature.

Thank you so much.  Alternatively, you may put a workaround for the key file and hardware key, when the user wants to use the system credentials to open the database.

Otherwise, for those who use both a password and a key, reopening the database with the key every few hours is the only option.

**Comment by J-Jamet** [comment]
> Not counter-intuitive for the user. The previous behaviour of the app wasn’t counter-intuitive. At least not for most users.

For many users, it was counterintuitive, otherwise I wouldn't have received these request at first.

> I don’t simply see the convention that you are trying to make as: ‘Swiping an app away MUST close the app, and MUST stop its functionality, and MUST clear it from the RAM.’.

I never said it was a convention. Personally, as long as KeePassDX is consistent in its behavior, that's enough. I just wanted to show you that what you're used to seeing and what you want isn't a universal standard with counterexamples. It's you who wants to make comparisons with other applications as if it were an argument to say that “others do it this way, so KeePassDX should too.”  You continue to do it, so that undoubtedly means that the issue is mainly a matter of habit.
And if the behavior was different before, you can imagine that it wasn't a fixed one from the outset and that it was subject to reflection and comparison.

As I mentioned earlier, the behavior changed for good reasons. It was a recurring request from users who didn't understand why their databases were still open after a swipe, and technically, in terms of KeePassDX's code, it just makes more sense. Previously, there was a setting to choose from the two behaviors, but it caused more problems than anything else because required completely different data management within the service.
I'm simply saying that for KeePassDX, it's the most sensible choice because one of the goals that sets it apart from other apps is to easily close a database without leaving any .kdbx data in memory.

> I don't think if an average user thinks of the Clear Button as a RAM Cleaner. Nothing about RAM and ‘RAM Cleaning’ is shown in the standard Recents/Overview screen.

Nothing prevents us from explaining it, even if it's only in the context of the application, by explaining, people will understand.

> Rather I think: ‘Swiping an app away MAY close the app, and MAY stop its functionality and, MAY clear it from the ram, IF it’s user’s desire!’

Yes, I agree.

> Now, at the moment, there is no consideration about ‘what the user may want!’.

I took many other opinions into consideration before yours and I take everything to make a choice. I can reconsider the matter, the proof is that we are discussing it. The app has been around since 2017 so there were many. If you disagree, I understand, but again, the behavior has changed for good reasons.

> As far as I know, closing the KeePassXC window through the top right close button or from the taskbar (or from the overview in Gnome) minimize it to tray icons (if the system have them) and keeps the program running in the background with the state of the database in this condition being controllable through an option in the preferences.

I installed a gnome specifically to test it, and no, KeePassXC closes with its database when the program's multi-view is displayed by placing the mouse in the upper left corner and the preview is closed with the top right close button.
Once again, this does not imply a convention but rather indicates another counterexample.

> One may have more than a hundred of apps installed, all of them are going to be used one day or the other, should all of them stay in the Recents/Overview screen?

Not necessarily, but there's no reason not to keep KeePassDX if you know that it meets your need to reuse an open database. 

> How many hours or minutes or days has to be the idle time distance between two sessions of an app being used, for it to stay in the Recents?

That's the beauty of it: it's up to the user to decide how long, since it's an action on their part.

> Who defines it?

The user himself

>  Is there an instruction about how to use the Overview screen?

If it's not clear, I can write a new page in the wiki for KeePassDX. It may just be a lack of information. This is an interesting topic. All other electronic devices that we buy in stores come with an instruction manual, but Android devices do not. If it is not that intuitive, there is nothing stopping us from writing a guide.

> The timer is not much useful if the app doesn’t stay open in the background. I may set it to eight hours, should I keep it in the Overview screen for eight hours? 

Yes, why not. 
The only thing that can prevent this is if:
- the system kills it, in which case it's an issue explained here https://dontkillmyapp.com/
- the user explicitly requests it with a swipe. If this is a problem for you, it's once again a matter of habit. Because if there is no user action, nothing happens, you just have a hidden preview screen that indicates that the database is open. If you know that, no worries.

> Isn’t it that the Overview has to contain apps that are used Recently. How many hours is recent? Is an app used yesterday considered Recent? If yes, then most of the installed apps has to stay in Overview screen.

Once again, it is up to the user to decide. If they set a time limit, then that determines the maximum time he gives to the definition of “recent” for an open database. If he decides he wants to close it earlier for any reason, just swipe or press the lock button and that's it.
For other applications, it is up to you to manage them according to your needs and their respective behavior.

> I have tried to use the credentials instead, but the KeePassDX doesn’t keep the key file. So, I need to re-provide the key file each time. What is the point of using the phone’s credentials then? If I’m going to provide the key file, I would simply insert the password, too, which is easier than providing the keyfile.

KeePassDX is fully capable of linking a key file URI to a database URI. `Settings - App settings - Remember keyfile locations`
A related feature : https://github.com/Kunzisoft/KeePassDX/issues/1315 

What I understand here is that you are promoting this feature as a workaround to keep the database open longer because you have no easy way to unlock it.
In this case, it makes more sense to address the source of your problem and make it easier to open your database rather than trying to keep it open. The key file can of course be easily reused, and the feature I mentioned will allow you to link the fingerprint to the key file.

> However, I use the right to disagree with your presumption about how the Overview screen SHOULD be used, as I described.

I understand, which is why I am explaining why the current behavior exists and under what circumstances I would reconsider reimplementing the feature.

> It doesn’t meet one of my needs, indeed. But still, it provides features not provided by other apps.

In this case, if other applications better suit your needs but lack functionality, Then the most logical thing to do is to add the features to the other application. I have no problem redirecting users to KeePass2Android. KeePassDX exists precisely because it has different ambitions; otherwise, the project would not even exist. But users who come from KeePass2Android want a clone, which is of no interest.

> Thank you so much. Alternatively, you may put a workaround for the key file and hardware key, when the user wants to use the system credentials to open the database.

We just need to focus on that but not as a workaround. If you have a related bug, feel free to open a new issue.

**Comment by curable-online** [comment]
> I never said it was a convention.

Well, I have to remind you that you said this:

> because the user is unequivocally asking to kill the application

And this:

> because killing the application with a swipe clearly means closing the database.

And this:

>  In this desired behavior, the user specifically wants to delete as much RAM memory as possible, so that is what is done.

And many other phrases which indicates your presumptions about how the user should be using the Recents screen.

> It's you who wants to make comparisons with other applications

I said:

> I can make a long list of apps that stay in the background doing their job after being swiped and a longer list of the apps that don’t.

Or, I said:

>  Now, about the Firefox, you are completely right. But, my example was trying to say this: “Swiping an app away doesn’t necessarily mean that it has to stop doing something a user wants it to do.”

And others, by which I tried to make example of other applications not assuming that “Swiping away an app from the Recents screen MUST kill the app and MUST stop its functionality, without exceptions.”.

The assumption which you sound like trying to make, as I quoted.

> “others do it this way, so KeePassDX should too.”

Did I? Did I say KeePassDX ‘should’ do, like anything?  But, I said:

> Swiping an app away just means I don't want to access it for now.

And this:

> Swiping away doesn't necessarily mean to stop an app's background activity. Many Android apps still do stuff in the background, even though they are not in Recents screen.

And this:

> an application can run in the background, and if it's user's desire, then it's the right thing to do so.

And this:

> Not wanting KeePassXD to be in my Recents screen, isn't semantically hard bound to my desire for my databases to be closed or open.

And this:

> ‘Swiping an app away MAY close the app, and MAY stop its functionality and, MAY clear it from the ram, IF it’s user’s desire!’

But, despite all these, let’s say you are right, and I’m the one who is trying to make assumptions about how a user should be using his/her Android device.  If that’s so, then I take the courage to take it back and restate:

“An application **can** run in the background, and if it's user's desire, then it's the right thing to do so.  Swiping away doesn't **necessarily** mean to stop an app's background activity.  Many Android apps still do stuff in the background, even though they are not in Recents screen.  Swiping an app away MAY close the app, and MAY stop its functionality and, MAY clear it from the ram, IF it’s user’s desire!”

Which is only repeating all what I already said before.  Which is not making assumptions, but breaking the assumptions you sound like trying to make.

> I installed a gnome specifically to test it, and no, KeePassXC closes with its database when the program's multi-view is displayed by placing the mouse in the upper left corner and the preview is closed with the top right close button.
> Once again, this does not imply a convention but rather indicates another counterexample.

<img width="1918" height="768" alt="Image" src="https://github.com/user-attachments/assets/c9a6b76a-8e68-4215-bf8d-6d8ae7107ecd" />

As you can see in the screenshot, they had these options since at least five years ago, when I started to use KeePassXC.  So, not a counterexample of what I don’t even know you was trying to counter.

> KeePassDX is fully capable of linking a key file URI to a database URI. Settings - App settings - Remember keyfile locations
> A related feature : #1315

When Keeping the keyfile of the database and database on the same storage?  It’s similar to having the password of your database in a plain-text file on the same storage as the database itself.  More like putting the key of the door, under the carpet in front of the door.  Well, that security measure worked, at least for my grandma.

The only **real** security advantage of a keyfile is to store it on a different place than the database itself.

And about keeping applications in the Recents menu, if one would keep all the applications she/he would use in eight hours in the Recents menu, then the Recents menu would hold just all the apps that have an icon in the home launcher (which includes most apps.).  That, makes Recents screen just a replication of the home launcher.  Don’t expect that from a user, please.  And, don’t think having users to keep an app in the Overview for hours for it to stay functioning as they would like is a beauty or freedom of choice.  It’s a lack of choice.

> What I understand here is that you are promoting this feature as a workaround to keep the database open longer because you have no easy way to unlock it.

It’s, indeed, understandable of a user to want to keep his/her database open, without clutter, without the risk of closing it by swipe.  Or, of a user to want to unlock his/her database using his/her system credentials, instead of using his/her other means of authorization, like hardware key, key file, or password.

> In this case, if other applications better suit your needs but lack functionality, Then the most logical thing to do is to add the features to the other application. I have no problem redirecting users to KeePass2Android.

Well, the right thing to do when an application is perfect, still
lacking a feature, is to add that lacking feature.

> But users who come from KeePass2Android want a clone, which is of no interest.

I’m not coming from KeePass2Android, I never installed it.  I heard its name ever in this thread.  Not only that, but I used KeePassDX on Android, since when I started using KeePass on Android.

**Comment by J-Jamet** [comment]
In the context of KeePassDX with the choices that have been made so far, yes, I still stand by what I said. No need to remind me, everything is valid.

> When Keeping the keyfile of the database and database on the same storage?

A URI does not take into account the storage location; the database and keyfile can be in two different locations, it's not a problem.
https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync

I never asked anyone to keep all their apps in the recent menu.

Just make the pull request with the setting if you want, and I'll look at it.
